feature-libs/quote/components/quote-seller-edit/quote-seller-edit.component.service.ts

```diff
@@ -0,0 +1,268 @@
+        //Symbol for currencies with symbol available, currency ISO code otherwise
+        const symbol = formatter
+          .formatToParts(0)
+          .find((x) => x.type === 'currency');
+        return this.checkAndReportCurrencyIfMissing(
+          locale,
+          formatter,
+          symbol?.value
+        );
+      })
+    );
+  }
+
+  /**
+   * Gets number format validator according to inputs
+   * @param locale Current locale
+   * @param currency Currency
+   * @param numberTotalPlaces Number of maximum total places
+   * @returns Formatter that can be attached to a form control
+   */
+  getNumberFormatValidator(
+    locale: string,
+    currency: string,
+    numberTotalPlaces: number
+  ): ValidatorFn {
+    return (control: AbstractControl): { [key: string]: any } | null => {
+      const input = control.value?.trim();
+      if (input) {
+        return this.getValidationErrorsNumericFormat(
+          input,
+          locale,
+          currency,
+          numberTotalPlaces
+        );
+      }
+      return null;
+    };
+  }
+
+  /**
+   * Adds current time and time zone to a date. Result is a timestamp string that can be handed over to OCC
+   * @param date Date as string
+   * @returns Timestamp as string
+   */
+  addTimeToDate(date: string): string {
+    const localTime = new Date().toLocaleTimeString([], {
+      hour: '2-digit',
+      minute: '2-digit',
+      hourCycle: 'h23',
+    });
+    return `${date}T${localTime}:00${TimeUtils.getLocalTimezoneOffset()}`;
+  }
+
+  /**
+   * Removes time portion from timestamp
+   * @param timestamp Timestamp as string
+   * @returns Date portion of timestamp
+   */
+  removeTimeFromDate(timestamp?: string): string | undefined {
+    return timestamp?.toString().substring(0, 10);
+  }
+
+  /**
+   * Check if quote state belongs to seller and can be edited
+   * @param quoteState
+   * @returns Is it for seller?
+   */
+  isEditableForSeller(quote: Quote): boolean {
+    const quoteState = quote.state;
+    return (
+      (quoteState === QuoteState.SELLER_REQUEST ||
+        quoteState === QuoteState.SELLER_DRAFT) &&
+      quote.isEditable
+    );
+  }
+
+  /**
+   * Get maximum number of decimal places. This supports validation, but it is not sufficient to do a complete validation,
+   * cases where the granted discount exceeds the total quote value by 1 are not covered (covered in OCC call).
+   * Still we want to inform the user as early as possible.
+   * Note that we assume currencies always come with 2 decimal places. In case this is not desired, this service can be overriden.
+   * @param quote Quote
+   * @returns Maximum number of places, including 2 decimal places
+   */
+  getMaximumNumberOfTotalPlaces(quote: Quote): number {
+    const numberOfDecimalPlaces = 2;
+    const maximum = Math.max(
+      quote.totalPrice.value ?? 1,
+      quote.quoteDiscounts?.value ?? 1
+    );
+    return Math.round(Math.log10(maximum) - 0.5) + 1 + numberOfDecimalPlaces;
```

Comment:
if you want to always round down, you may use Math.floor
```suggestion
    return Math.floor(Math.log10(maximum)) + 1 + numberOfDecimalPlaces;
```

Reply:
Agree, good point